android-animation-arsenal/src/main/java/com/android_animation_arsenal/views/RevealDialog.java

```diff
@@ -0,0 +1,373 @@
+                        revealDialogWithTarget();
+                    }
+                    else
+                    {
+                        revealDialog();
+                    }
+                }
+            });
+        }
+        super.show();
+    }
+
+    /**
+     * Reveals dialog without target
+     */
+    public void revealDialog()
+    {
+        int x = contentView.getWidth() / 2;
+        int y = contentView.getHeight() / 2;
+        DisplayMetrics displayMetrics = getContext().getResources().getDisplayMetrics();
+        // big radius to cover view
+        int bigRadius = Math.max(displayMetrics.widthPixels, displayMetrics
+                .heightPixels);
+        Animator toolbarExpandAnim = ViewAnimationUtils.createCircularReveal(
+                contentView, x, y, 0, bigRadius);
+        toolbarExpandAnim.setStartDelay(duration);
+        toolbarExpandAnim.setDuration(duration);
+    }
+
+    @Override
+    public void setContentView(View view)
+    {
+        super.setContentView(view);
+        contentView = view;
+        contentView.setOnTouchListener(new View.OnTouchListener()
+        {
+            @Override
+            public boolean onTouch(View v, MotionEvent event)
+            {
+                return true;
+            }
+        });
+    }
+
+    /**
+     * Animates dialog into a reveal and handles
+     * target animation from its original position to dialog position
+     */
+    public void revealDialogWithTarget()
+    {
+        if(isAnimating)
+        {
+            return;
+        }
+
+        isAnimating = true;
+
+        // get display metrics
+        DisplayMetrics metrics = new DisplayMetrics();
+        getWindow().getWindowManager().getDefaultDisplay().getMetrics(metrics);
+
+        // get center coordinates of dialog
+        final int x = contentView.getWidth() / 2;
+        final int y = contentView.getHeight() / 2;
+
+        // disable targetView
+        targetView.setEnabled(false);
+        targetView.setClickable(false);
+
+        // change alpha of contentView and visibility
+        contentView.setAlpha(0f);
+        contentView.setVisibility(View.VISIBLE);
+
+        // get original position of targetView
+        final int originalPos[] = new int[2];
+        targetView.getLocationOnScreen(originalPos);
+
+        // set x and y destination for targetView
+        final int xDest = (metrics.widthPixels / 2) - (targetView.getMeasuredWidth() / 2);
+        final int yDest = (metrics.heightPixels / 2) - (targetView.getMeasuredHeight() / 2);
+
+        // animate targetView Button x transition
+        Animator targetSlideXAnim = ObjectAnimator.ofPropertyValuesHolder(targetView,
+                PropertyValuesHolder.ofFloat("translationX", 0f, xDest -
+                        originalPos[0]));
+        targetSlideXAnim.setDuration(duration);
+
+        // animate targetView y transition
+        Animator targetSlideYAnim = ObjectAnimator.ofPropertyValuesHolder(targetView,
```

Comment:
Will be a good idea to also create functions to return X and Y Property animators ?